From bd321e3c176604957deef59554a565d6db76584b Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Sat, 30 Aug 2025 00:37:42 +0200
Subject: [PATCH] Nuke Knox SDP

UN1CA: services.jar already carries the current-base SDP stubs. Keep this
patch limited to the remaining Secure Folder SDP support check; the framework
side no longer calls the binder helper, but direct service calls must also
return false.
---
 smali/com/android/server/knox/dar/DarManagerService.smali | 55 ---------------
 1 file changed, 55 deletions(-)

diff --git a/smali/com/android/server/knox/dar/DarManagerService.smali b/smali/com/android/server/knox/dar/DarManagerService.smali
--- a/smali/com/android/server/knox/dar/DarManagerService.smali
+++ b/smali/com/android/server/knox/dar/DarManagerService.smali
@@ -2429,62 +2429,7 @@
 .method public final isSdpSupportedSecureFolder(I)Z
     .locals 3
 
-    iget-object v0, p0, Lcom/android/server/knox/dar/DarManagerService;->mInjector:Lcom/android/server/knox/dar/DarManagerService$Injector;
-
-    invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
-
-    invoke-static {}, Landroid/os/Binder;->clearCallingIdentity()J
-
-    move-result-wide v0
-
-    iget-object v2, p0, Lcom/android/server/knox/dar/DarManagerService;->mUserManager:Landroid/os/UserManager;
-
-    if-eqz v2, :cond_0
-
-    invoke-virtual {v2, p1}, Landroid/os/UserManager;->getUserInfo(I)Landroid/content/pm/UserInfo;
-
-    move-result-object p1
-
-    goto :goto_0
-
-    :cond_0
-    const/4 p1, 0x0
-
-    :goto_0
-    iget-object p0, p0, Lcom/android/server/knox/dar/DarManagerService;->mInjector:Lcom/android/server/knox/dar/DarManagerService$Injector;
-
-    invoke-virtual {p0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
-
-    invoke-static {v0, v1}, Landroid/os/Binder;->restoreCallingIdentity(J)V
-
-    if-eqz p1, :cond_1
-
-    goto :goto_1
-
-    :cond_1
-    sget-object p1, Lcom/android/server/knox/dar/DarManagerService;->NULL_USER:Landroid/content/pm/UserInfo;
-
-    :goto_1
-    if-eqz p1, :cond_2
-
-    invoke-virtual {p1}, Landroid/content/pm/UserInfo;->isSecureFolder()Z
-
-    move-result p0
-
-    if-eqz p0, :cond_2
-
-    invoke-virtual {p1}, Landroid/content/pm/UserInfo;->isSdpNotSupportedSecureFolder()Z
-
-    move-result p0
-
-    if-nez p0, :cond_2
-
-    const/4 p0, 0x1
-
-    return p0
-
-    :cond_2
     const/4 p0, 0x0
 
     return p0
 .end method
-- 
2.51.1
